server/util/strategicLogger.ts

```diff
@@ -4,9 +4,9 @@ const pgPromise = require('pg-promise');
 const octokit = require('@octokit/rest')();
 
 function sleep() {
-  /* Use random sleep time (3-12 seconds) to avoid stampeding the API */
+  /* Use random sleep time (3-7 seconds) to avoid stampeding the API */
   return new Promise(resolve =>
-    setTimeout(resolve, Math.floor(Math.random() * 10000 + 3000))
+    setTimeout(resolve, Math.floor(Math.random() * 4000 + 3000))
```

Comment:
You can probably be significantly more aggressive than this, but use what works.

Reply:
Anything less than 3 seconds causes problems but this is much tighter than it was.